lib/dao.js

```diff
@@ -3059,7 +3059,10 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
 
         function replaceCallback(err, data) {
           if (err) return cb(err);
-
+          // if connector is cloudant, add the `_rev` property back before passing it to the user
+          if (connector.name === 'cloudant') {
+            context.data._rev = data._rev;
```

Comment:
agree with @b-admike I have been confused by this for long time: those CRUD functions return `data` that user passes in, instead of the real data returned from request to database....Not a block of this PR, just a general question :p